submissions/mike-rashid/scripts/article.js

```diff
@@ -85,3 +85,38 @@ Article.fetchAll = function() {
   } else {}
 }
 */
+
+Article.fetchAll = function (){
+  if (localStorage.hackerIpsum){
+    var newEtag;
+    $.ajax({
+      url: '/data/hackerIpsum.json',
```

Comment:
This is a pretty heavy AJAX call when you just want the eTag. Try just getting the headers (it's totally possible). 

Though if you keep it like this, you don't need to make a separate AJAX call for the data if it doesn't match. You already have the data.